src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService_UpdateSource.cs

```diff
@@ -40,41 +40,59 @@ internal partial class DiagnosticAnalyzerService : IDiagnosticUpdateSource
             }
         }
 
-        internal void RaiseDiagnosticsUpdated(DiagnosticsUpdatedArgs args)
+        internal void RaiseDiagnosticsUpdated(ImmutableArray<DiagnosticsUpdatedArgs> args)
         {
             // all diagnostics events are serialized.
-            var ev = _eventMap.GetEventHandlers<EventHandler<DiagnosticsUpdatedArgs>>(DiagnosticsUpdatedEventName);
+            var ev = _eventMap.GetEventHandlers<EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>>(DiagnosticsUpdatedEventName);
             if (ev.HasHandlers)
             {
                 _eventQueue.ScheduleTask(nameof(RaiseDiagnosticsUpdated), () => ev.RaiseEvent(static (handler, arg) => handler(arg.self, arg.args), (self: this, args)), CancellationToken.None);
             }
         }
 
-        internal void RaiseBulkDiagnosticsUpdated(Action<Action<DiagnosticsUpdatedArgs>> eventAction)
+        internal void RaiseBulkDiagnosticsUpdated(Action<Action<ImmutableArray<DiagnosticsUpdatedArgs>>> eventAction)
         {
             // all diagnostics events are serialized.
-            var ev = _eventMap.GetEventHandlers<EventHandler<DiagnosticsUpdatedArgs>>(DiagnosticsUpdatedEventName);
+            var ev = _eventMap.GetEventHandlers<EventHandler<ImmutableArray<DiagnosticsUpdatedArgs>>>(DiagnosticsUpdatedEventName);
             if (ev.HasHandlers)
             {
                 // we do this bulk update to reduce number of tasks (with captured data) enqueued.
                 // we saw some "out of memory" due to us having long list of pending tasks in memory. 
                 // this is to reduce for such case to happen.
-                void raiseEvents(DiagnosticsUpdatedArgs args) => ev.RaiseEvent(static (handler, arg) => handler(arg.self, arg.args), (self: this, args));
+                void raiseEvents(ImmutableArray<DiagnosticsUpdatedArgs> args)
+                {
+                    ev.RaiseEvent(
+                        static (handler, arg) =>
+                        {
+                            if (!arg.args.IsEmpty)
```

Comment:
>if (!arg.args.IsEmpty) [](http://example.com/codeflow?start=28&length=22)

nit: could be moved outside the RaiseEvent call to prevent the need to call it

Reply:
⏱️ Updating this one

Reply:
➡️ Now updated